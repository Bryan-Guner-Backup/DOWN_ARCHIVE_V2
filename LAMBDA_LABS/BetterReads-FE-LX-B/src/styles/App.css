.bookContainer{
    display: flex;
}

.bookContainer .bookThumbStatus {
    display: flex;
    flex-direction: column;
}

.bookContainer .bookDetails{
    display: flex;
    flex-direction: column;   
}

.bookContainer .bookDetails .bookTitleFavIcon{
    display: flex;
    flex-direction: row;
}