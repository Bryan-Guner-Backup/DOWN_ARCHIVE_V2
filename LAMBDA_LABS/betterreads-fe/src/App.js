import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import { PrivateRoute } from './utils/PrivateRoute';
// Components
import Library from './components/library/Library';
import SignIn from './components/authentication/SignIn';
import SignUp from './components/authentication/SignUp';
import PassReset from './components/authentication/PassReset';
import Success from './components/authentication/Success';
import Failure from './components/authentication/Failure';
import PageNotFound from './components/authentication/PageNotFound';
import Search from './components/search/Search';
import ShelfBook from './components/library/ShelfBook';
import Shelf from './components/library/Shelf';
import Shelves from './components/library/Shelves';
import OnboardingQuiz from './components/authentication/OnboardingQuiz';
import LandingPage from './components/Landing/Landing';
// History Util
import history from './utils/history';

const App = () => {
  return (
    <Router history={history}>
      <Switch>
        <PrivateRoute exact path='/home' component={Library} />
        <Route exact path='/' component={LandingPage} />
        <Route exact path='/signin' component={SignIn} />
        <Route exact path='/signup' component={SignUp} />
        <Route exact path='/reset' component={PassReset} />
        <Route exact path='/quiz' component={OnboardingQuiz} />
        <Route exact path='/success' component={Success} />
        <Route exact path='/failure' component={Failure} />
        <Route exact path='/pagenotfound' component={PageNotFound} />
        <PrivateRoute exact path='/search' component={Search} />
        <PrivateRoute exact path='/book/:id' component={ShelfBook} />
        <PrivateRoute exact path='/shelf/:shelf' component={Shelf} />
        <PrivateRoute exact path='/myshelves' component={Shelves} />
        <Route component={PageNotFound} />
      </Switch>
    </Router>
  );
};

export default App;
