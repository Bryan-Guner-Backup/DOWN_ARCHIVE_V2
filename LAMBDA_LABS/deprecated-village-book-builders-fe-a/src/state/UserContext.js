import React, { useState, useContext, useEffect } from 'react';

const UserContext = React.createContext();

export function useUser() {
  return useContext(UserContext);
}

export function UserProvider({ children }) {
  const [userInfo, setUserInfo] = useState({
    username: localStorage.getItem('username'),
    role: localStorage.getItem('role'),
    id: localStorage.getItem('id'),
  });

  useEffect(() => {
    if (
      !localStorage.getItem('username') === null ||
      !localStorage.getItem('role') === null
    ) {
      return;
    } else {
      localStorage.setItem('username', userInfo.username);
      localStorage.setItem('role', userInfo.role);
      localStorage.setItem('id', userInfo.id);
    }
  }, [userInfo]);
  return (
    <UserContext.Provider value={{ userInfo, setUserInfo }}>
      {children}
    </UserContext.Provider>
  );
}
