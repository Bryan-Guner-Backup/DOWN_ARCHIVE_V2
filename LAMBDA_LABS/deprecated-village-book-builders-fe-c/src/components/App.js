import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import '../style.css';
// import PrivateRoute from "../utils/PrivateRoute";
import { checkToken } from '../state/actions/index';
import Login from './pages/Login/Login';
import HeadmasterDashboard from './pages/Headmaster/HeadmasterDashboard';
import Profile from '../components/pages/Headmaster/HeadmasterProfile/Profile';
import ProfileForm from '../components/pages/Headmaster/HeadmasterProfile/ProfileForm';
import AdminDashboard from './pages/Admin/AdminDashboard';
import TeacherProfile from './pages/Teacher/TeacherProfile';
import TeacherEditProfile from './pages/Teacher/TeacherEditProfile';
import StudentDashboard from './pages/Student/StudentDashboard';
import MenteeDashboard from './pages/Headmaster/Mentees/MenteeDashboard';
import StudentOnboarding from './pages/Student/StudentOnboarding';

const App = ({ role, checkToken }) => {
  return (
    <div className="App">
      <Switch>
        <Route path="/login">
          <Login />
        </Route>
        <Route path="/headmaster/edit">
          <ProfileForm />
        </Route>
        <Route path="/headmaster">
          <Profile />
        </Route>
        {/*Teacher Routes*/}
        <Route path="/teacher/edit/:id">
          <TeacherEditProfile />
        </Route>
        <Route path="/teacher/:id">
          <TeacherProfile />
        </Route>
        <Route path="/student-onboarding">
          <StudentOnboarding />
        </Route>

        <Route path="/">
          {/*//! this needs to be changed to if there is an unexpired token*/}
          {/* Look for token in case a user refreshes the page & clears redux store, then it repopulates the redux store with userId, role & loggedIn status with checkToken().  */}
          {localStorage.getItem('token') ? (
            <>
              {checkToken()}
              {/* //once we make a reusable dashboard/sidebar, this is where we would put it, passing in the role as props to fill it out accordingly. */}
              {role === 'headmaster' && <HeadmasterDashboard />}
              {role === 'admin' && <AdminDashboard />}

              {/* checks token to see if role is mentee. credentials to check with -> mentees@mentees.com : password */}
              {role === 'student' && <StudentDashboard />}
              {role === 'mentee' && <MenteeDashboard />}
            </>
          ) : (
            <Redirect to="/login" />
          )}
        </Route>
      </Switch>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    // loggedIn: state.authReducer.loggedIn,
    // userId: state.authReducer.userId,
    role: state.authReducer.role,
  };
};

export default connect(mapStateToProps, { checkToken })(App);
