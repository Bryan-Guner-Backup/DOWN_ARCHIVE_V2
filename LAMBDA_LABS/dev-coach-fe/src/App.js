import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  createMuiTheme,
  ThemeProvider,
} from '@material-ui/core/styles';
import LoginForm from './components/Onboarding/Login/LoginForm';
import ResetPasswordForm from './components/Onboarding/ResetPasswordForm';
import AccountRecovery from './components/Onboarding/AccountRecovery';
import Dashboard from './components/Dashboard';
import UserDashboard from './views/UserDashboard/UserDashboard';
import Marketplace from './views/Marketplace/Marketplace';
import Landing from './components/Landing/Landing-2';
import MainFaq from './components/FAQ/Main';
import LandingFaq from './components/FAQ/LandingFaq';
import Booking from './components/Booking/Booking';
import Reschedule from './components/Reschedule/Reschedule';
import Feedback from './views/Feedback/Feedback';
import Interview from './views/Interview/Interview';
import Chat from './components/Chat/ChatScreen';
import StartChat from './components/Chat/Chat';
import SignUp from './components/Onboarding/SignupStepper';
import SettingsTabs from './views/Settings/SettingsTabs';
import Code from './views/Code/Code';
import ResourceHub from './views/ResourceHub/ResourceHub';
import GiveFeedback from './views/Feedback/GiveFeedback';
import About from './components/About/About';
import Team from "./components/Team/Team"

const globalTheme = createMuiTheme({
  typography: {
    fontFamily: ['Nunito', 'Helvetica', 'sans-serif'].join(','),
  },
});

function App({ user, isLoggedIn }) {
  const url = process.env.REACT_APP_BASE_URL;
  const routes = (
    <Switch>
      <Route exact path='/' component={Landing} />
      <Route path={'/dashboard'} component={UserDashboard} />
      {user && user.role_id === 1 ? (
        <Route path={'/marketplace'} component={Marketplace} />
      ) : null}
      <Route path={'/resourcehub'} component={ResourceHub} />
      <Route path={'/appointment'} component={Booking} />
      <Route path={'/reschedule'} component={Reschedule} />
      <Route path={'/feedback'} component={Feedback} />
      <Route path={'/interview'} component={Interview} />
      <Route path={'/Settings'} component={SettingsTabs} />
      <Route path={'/FAQ'} component={MainFaq} />
      <Route path={'/start_chat'} component={StartChat} />
      <Route path={'/chat'} component={Chat} />
      <Route path={'/code'} component={Code} />
      <Route path={'/givefeedback'} component={GiveFeedback} />
      <Redirect to='/dashboard' />
    </Switch>
  );

  if (isLoggedIn) {
    return (
      <ThemeProvider theme={globalTheme}>
        <Switch>
          <Route exact path='/' component={Landing} />
          <Route path='/about' component={About} />
          <Dashboard routes={routes} />
        </Switch>
      </ThemeProvider>
    );
  }

  return (
    <Switch>
      <Route exact path='/' component={Landing} />
      <Route path='/login/' component={LoginForm} />
      <Route path='/resetPassword' component={ResetPasswordForm} />
      <Route
        path='/accountRecovery/:token'
        component={AccountRecovery}
      />
      <Route
        path='/register'
        render={props => <SignUp url={url} {...props} />}
      />
      <Route path='/faq' component={LandingFaq} />
      <Route path='/about' component={About} />
      <Route path={'/video'} component={Interview} />
      <Route path ={"/team"} component={Team} />
      <Redirect to='/' />
    </Switch>
  );
}

const mapStateToProps = state => {
  return {
    user: state.userReducer.user,
    isLoggedIn: state.userReducer.isLoggedIn,
  };
};

export default connect(mapStateToProps)(App);
