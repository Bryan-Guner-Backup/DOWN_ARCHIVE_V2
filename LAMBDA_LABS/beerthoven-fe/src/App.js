import React from "react";
import { Route, Switch } from "react-router-dom";
import "./App.css";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { Layout, Menu } from "antd";
import { Security, SecureRoute, LoginCallback } from "@okta/okta-react";
import oktaConfig from "./okta-config";
import OktaApolloProvider from "./components/utils/OktaApolloProvider";
import LogoutButton from "./components/LogoutButton";
import Dashboard from "./components/pages/Dashboard";
import Login from "./components/pages/Login";
import People from "./components/pages/People";
import Event from "./components/pages/Event";
import Venue from "./components/pages/Venue";
import User from "./components/pages/User";
const { Sider, Content } = Layout;

function App() {
  return (
    <div className="App" data-testid="app">
      <Router>
        <Security {...oktaConfig}>
          <OktaApolloProvider>
            <Layout>
              <Sider
                style={{
                  overflow: "auto",
                  height: "100vh",
                  position: "fixed",
                  left: 0,
                }}
              >
                <Menu theme="dark">
                  <Menu.ItemGroup key="h1" title="Manage">
                    <Menu.Item key="1">
                      <Link to="/">Dashboard</Link>
                    </Menu.Item>
                    <Menu.Item key="2">
                      <Link to="/people">People</Link>
                    </Menu.Item>
                    <Menu.Item key="3">
                      <Link to="/event">Events</Link>
                    </Menu.Item>
                    <Menu.Item key="4">
                      <Link to="/venue">Venues</Link>
                    </Menu.Item>
                    <Menu.Item key="5">
                      <Link to="/user">Users</Link>
                    </Menu.Item>
                  </Menu.ItemGroup>
                </Menu>
                <div
                  style={{
                    paddingTop: "30px",
                    display: "grid",
                    placeItems: "center",
                  }}
                >
                  <LogoutButton />
                </div>
              </Sider>
              <Layout
                style={{
                  marginLeft: 200,
                  height: "100vh",
                }}
              >
                <Content style={{ margin: "24px 16px 0", overflow: "initial" }}>
                  <Switch>
                    <SecureRoute path="/login" component={Login} />
                    <SecureRoute path="/people" component={People} />
                    <SecureRoute path="/event" component={Event} />
                    <SecureRoute path="/venue" component={Venue} />
                    <SecureRoute path="/user" component={User} />
                    <Route
                      path={oktaConfig.callbackPath}
                      component={LoginCallback}
                    />
                    <SecureRoute exact path="/" component={Dashboard} />
                  </Switch>
                </Content>
              </Layout>
            </Layout>
          </OktaApolloProvider>
        </Security>
      </Router>
    </div>
  );
}

export default App;
