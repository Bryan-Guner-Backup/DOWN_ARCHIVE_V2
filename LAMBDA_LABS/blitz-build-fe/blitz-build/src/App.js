//notes
import React, { useState, useEffect } from "react";
import { Switch } from "react-router";
import { BrowserRouter as Router, Route } from "react-router-dom";
// import Login from "./components/auth/Login";
// import Signup from "./components/auth/Signup";
// import PrivateRoute from "./components/auth/PrivateRoute";
import NavBar from "./components/NavBar";
import jwtDecode from "jwt-decode";
// import Layout from "./components/dashboard/Layout";
// import Dashboard from "./components/dashboard/index";
// import Dashboard from "./components/dashboard/Dashboard";

//task views
import Tasks from "./views/tasks/Tasks";
import ProjectTasks from "./views/tasks/ProjectTasks";

import Templates from "./components/templates/templates";
import IndividualTemplate from "./components/templates/IndividualTemplate";
import NinetyDayTemplate from "./components/templates/90DayTemplate";
import Projects from "./views/projects/Projects";
import IndividualProject from "./views/projects/IndividualProject";
import Logout from "./components/auth/Logout";
import Layout from "./layouts/Layout";
import Dashboard from "./components/dashboard/Dashboard";
import DelayLog from "./views/delayLog/DelayLog";
import OpenTemplateContext from "./contexts/OpenTemplateContext";

import Documents from "./components/documents/Documents";
//SWITCH INDEX TO DASHBOARD AFTER LC CHANGES HIS FILE NAME
import ActivityViewAll from "./components/activityFeed/ActivityViewAll";

import Landing from "./components/landing/pages/Landing";

//context
import UserContext from "./contexts/UserContext";
import TaskProvider from "./contexts/tasks/TaskProvider";
import DelayLogProvider from "./contexts/delayLog/DelayLogProvider";
import PathnameContext from "./contexts/PathnameContext";
import ProjectsProvider from "./contexts/projects/ProjectsProvider";
import EditModalContext from "./contexts/EditModalContext";
import SearchProvider from "./contexts/searching/searchTermProvider";
// import SearchTermContext from "./contexts/searching/searchTerm";
// import TemplateProvider from "./contexts/templates/TemplateProvider";
// import { axiosWithAuth } from "./utils/auth/axiosWithAuth";
import ComponentToPrint from "./components/documents/PrintDocument";
//AUTH0
import Auth from "./components/auth/auth";
import AuthNavBar from "./components/auth/authNavBar";
import Callback from "./components/auth/callback";
import ViewDocument from "./components/documents/ViewDocument";
import TemplatesProvider from "./contexts/templates/TemplateProvider";
import Uploader from "./components/documents/Uploader";
import MyCalendar from "./components/calendar/MyCalender";
import DocumentsProvider from "./contexts/documents/DocumentsProvider";
function App() {
  const [pathname, setPathname] = useState(window.location.pathname);
  const [openTemplate, setOpenTemplate] = useState(false);
  const [userInfo, setUserInfo] = useState([]);
  const [editModalOpen, setEditModalOpen] = useState(false);

  // getting userInfo from id_token in localStorage.
  useEffect(() => {
    if (localStorage.getItem("id_token")) {
      setUserInfo(jwtDecode(localStorage.getItem("id_token")));
    }
  }, []);

  //function that checks the path to see if it should return the marketing page
  const LandingPage = component => {
    if (pathname === "/") {
      return null;
    } else {
      return component;
    }
  };

  // console.log("userInfo", userInfo);
  console.log("pathname", pathname);
  const navLinks = [
    {
      text: "Home",
      path: "/dashboard",
      icon: "ion-ios-home"
    },
    {
      text: "Projects",
      path: "/projects",
      icon: "ion-ios-construct"
    },
    {
      text: "Tasks",
      path: "/tasks?filter=ACTIVE",
      icon: "ion-ios-notifications"
    },
    {
      text: "Documents",
      path: "/documents",
      icon: "ion-ios-document"
    },
    {
      text: "Templates",
      path: "/templates",
      icon: "ion-ios-menu"
    },
    {
      text: "Delay Log",
      path: "/delay-log",
      icon: "ion-ios-hourglass"
    },
    {
      text: "MyCalendar",
      path: "/mycalendar",
      icon: "ion-ios-calendar"
    },
    {
      text: "Log Out",
      path: "/log-out",
      icon: "ion-ios-cog"
    },
    {
      text: "Help",
      path: "/help",
      icon: "ion-ios-help-circle-outline"
    }
  ];

  return (
    <Router>
      <TemplatesProvider>
        <OpenTemplateContext.Provider value={{ openTemplate, setOpenTemplate }}>
          <ProjectsProvider>
            <DocumentsProvider>
              <TaskProvider>
                <DelayLogProvider>
                  <SearchProvider>
                    <EditModalContext.Provider
                      value={{ editModalOpen, setEditModalOpen }}
                    >
                      <PathnameContext.Provider
                        value={{ pathname, setPathname }}
                      >
                        <UserContext.Provider value={{ userInfo, setUserInfo }}>
                          {LandingPage(
                            <NavBar
                              setPathname={setPathname}
                              navLinks={navLinks}
                            />
                          )}
                          <Layout pathname={pathname}>
                            <Switch>
                              <Route exact path="/" component={Landing} />

                              <Route
                                exact
                                path="/navbar"
                                component={AuthNavBar}
                              />
                              <Route
                                exact
                                path="/callback"
                                component={Callback}
                              />
                              <Route
                                exact
                                path="/activity-feed"
                                component={ActivityViewAll}
                              />
                              <Route
                                exact
                                path="/login"
                                // component={Login}
                              />

                              {/* <Route exact path="/signup" component={Signup} /> */}
                              <Route exact path="/log-out" component={Logout} />
                              {/*   */}
                              <Route
                                exact
                                path="/dashboard"
                                component={Dashboard}
                              />
                              <Route exact path="/tasks" component={Tasks} />
                              <Route
                                exact
                                path="/projects"
                                component={Projects}
                              />
                              <Route
                                exact
                                path="/projects/:id"
                                component={IndividualProject}
                              />
                              <Route
                                exact
                                path="/projects/:id/tasks"
                                component={ProjectTasks}
                              />
                              <Route
                                exact
                                path="/templates"
                                component={Templates}
                              />
                              <Route
                                exact
                                path="/templates/:id"
                                component={IndividualTemplate}
                              />
                              <Route
                                exact
                                path="/90_Day"
                                component={NinetyDayTemplate}
                              />
                              <Route
                                exact
                                path="/delay-log"
                                component={DelayLog}
                              />
                              <Route
                                exact
                                path="/documents"
                                component={Documents}
                              />
                              <Route
                                exact
                                path="/documents/add"
                                component={Uploader}
                              />
                              <Route
                                exact
                                path="/documents/view/:file_name"
                                component={ViewDocument}
                              />
                              <Route
                                exact
                                path="/documents/print"
                                component={ComponentToPrint}
                              />
                              <Route
                                exact
                                path="/mycalendar"
                                component={MyCalendar}
                              />
                            </Switch>
                          </Layout>
                        </UserContext.Provider>
                      </PathnameContext.Provider>
                    </EditModalContext.Provider>
                  </SearchProvider>
                </DelayLogProvider>
              </TaskProvider>
            </DocumentsProvider>
          </ProjectsProvider>
        </OpenTemplateContext.Provider>
      </TemplatesProvider>
    </Router>
  );
}

export default App;
