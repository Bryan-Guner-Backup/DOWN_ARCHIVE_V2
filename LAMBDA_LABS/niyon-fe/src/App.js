import React, { useEffect, useState } from 'react'
import { UserContext } from './UserContext'
import { DarkModeContext } from './DarkModeContext'
import Routes from './components/routes/Routes'
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles'
import './sass_master/index.scss'
import CssBaseline from '@material-ui/core/CssBaseline'

function App () {
  const [user, setUser] = useState({})
  const [darkMode, setDarkMode] = useState(false)

  useEffect(() => {
    if (darkMode === true) {
      document.body.classList.add('dark-mode')
    } else {
      document.body.classList.remove('dark-mode')
    }
  }, [darkMode])

  const theme = React.useMemo(
    () =>
      createMuiTheme({
        palette: {
          type: darkMode ? 'dark' : 'light'
        }
      }),
    [darkMode]
  )

  return (
    <UserContext.Provider value={{ user, setUser }}>
      <DarkModeContext.Provider value={{ darkMode, setDarkMode }}>
        <ThemeProvider theme={theme}>
          <CssBaseline />
          <Routes />
        </ThemeProvider>
      </DarkModeContext.Provider>
    </UserContext.Provider>
  )
}

export default App
