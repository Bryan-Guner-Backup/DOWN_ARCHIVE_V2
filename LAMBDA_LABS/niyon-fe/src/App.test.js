import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';
import  { UserContext } from './UserContext';
import findByTestAttr from './tests/utils/index';


describe('<App /> Component', () => {
     const user = {};
     const mode=false;
     const setMode = jest.fn()
     const setUser = jest.fn(); 
     const setUserSpy = jest.spyOn(React, 'useState');
     setUserSpy.mockImplementation((user) => [user, setUser]);
     let component;
     let useEffect;
     const mockUseEffect = () => {
          useEffect.mockImplementationOnce(() => setMode());
     };
     beforeEach(() => {
     useEffect = jest.spyOn(React, "useEffect");
     mockUseEffect(); // 2 times
     mockUseEffect(); //    
     component = shallow(        
            <UserContext.Provider value={{user, setUser}}>
              <App />
            </UserContext.Provider>       
          );      
     });
     
      it('should match the snapshot testing', () => {
           expect(toJSON(component)).toMatchSnapshot();
      })

     it('should render <App /> component correctly', () => {
           expect(component.exists()).toBe(true);
     });
     

     it('should render <Routes /> component correctly', () => {
          const wrapper = component.dive();          
          expect(wrapper.find('Routes').exists()).toBe(true);
     });    
});