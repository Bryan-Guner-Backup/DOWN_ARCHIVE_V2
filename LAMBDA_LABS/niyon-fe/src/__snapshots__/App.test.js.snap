// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component should match the snapshot testing 1`] = `<App />`;
